ui/src/app/base/components/MainTable/MainTable.js

```diff
@@ -0,0 +1,209 @@
+import PropTypes from "prop-types";
+import React, { useEffect, useState } from "react";
+
+import Table from "../Table";
+import TableRow from "../TableRow";
+import TableHeader from "../TableHeader";
+import TableCell from "../TableCell";
+
+const _updateSort = (
+  _setSortKey,
+  _setSortDirection,
+  sortKey,
+  sortDirection
+) => {
+  let newDirection = null;
+  if (sortDirection === "none") {
+    newDirection = "ascending";
+  } else if (sortDirection === "ascending") {
+    newDirection = "descending";
+  } else {
+    sortKey = null;
+  }
+  _setSortKey(sortKey);
+  _setSortDirection(newDirection);
+};
+
+const _generateHeaders = (
```

Comment:
Small thing, but as these aren't class methods, and these functions aren't exported, I'm not sure the `_` prefix is meaningful now? Not overly important though.

Reply:
Done.